src/views/webviewController.ts

```diff
@@ -105,6 +106,54 @@ export default class WebviewController {
     this._themeChangedSubscription?.dispose();
   }
 
+  handleWebviewOpenFileChooserAttempt = async ({
+    panel,
+    fileChooserOptions,
+    requestId,
+  }: {
+    panel: vscode.WebviewPanel;
+    fileChooserOptions: FileChooserOptions;
+    requestId: string;
+  }): Promise<void> => {
+    let files;
+
+    try {
+      files = await vscode.window[
+        fileChooserOptions.mode === 'open' ? 'showOpenDialog' : 'showSaveDialog'
```

Comment:
Since `mode` is an enum, these kinds of if-statements are very brittle as they assume that it will only ever have two values. Instead, it's better to use a switch, even if it means the code will be a bit more verbose.